### PR TITLE
2SV email notifications

### PR DESCRIPTION
**Pull request by benlovell**
This adds various user email notifications around the 2SV actions.

### CODE REVIEW

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -6,6 +6,26 @@ class UserMailer < Devise::Mailer
 
   helper_method :suspension_time, :account_name, :instance_name, :locked_time, :unlock_time
 
+  def two_step_reset(user)
+    @user = user
+    mail(to: @user.email, subject: "2-step verification is now reset")
```
Comment (fofr) [inline app/mailers/user_mailer.rb]: Would this be better:
"Your 2-step verification has been reset"

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -6,6 +6,26 @@ class UserMailer < Devise::Mailer
...
+  end
+
+  def two_step_enabled(user)
+    @user = user
+    mail(to: @user.email, subject: "2-step verification set up")
+  end
+
+  def two_step_flagged(user)
+    @user = user
+    mail(to: @user.email, subject: "Set up 2-step verification")
```
Comment (fofr) [inline app/mailers/user_mailer.rb]: When we prompt users when they Sign in we use the title "Make your account more secure", can we do the same here?

Comment (benlovell) [inline app/mailers/user_mailer.rb]: I should have mentioned, like the body, the subjects are WIP placeholders for now. I'd assume the content team provides both subject and body? Happy to go with your suggestions though if they're definitive. :+1:

File: test/integration/super_admin_reset_two_step_verification_test.rb [hunk 3]
```diff
@@ -27,12 +30,19 @@ class SuperAdminResetTwoStepVerificationTest < ActionDispatch::IntegrationTest
...
+        assert_response_contains '2-step verification enabled'
+
+        click_link 'Reset 2-step verification'
 
-      click_link 'Reset 2-step verification'
+        assert_response_contains MESSAGE
 
-      assert_response_contains('2-step verification is now reset')
+        assert last_email
+        assert_equal '2-step verification is now reset', last_email.subject
```
Comment (benlovell) [inline test/integration/super_admin_reset_two_step_verification_test.rb]: Gah.

File: test/integration/super_admin_flagging_two_step_verification_test.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          assert last_email
+          assert_equal 'Set up 2-step verification', last_email.subject
+        end
+      end
+    end
+
+    context 'when the user is not flagged for 2SV' do
+      should 'not notify the user by email' do
+        perform_enqueued_jobs do
+          uncheck 'Ask user to set up 2-step verification'
```
Comment (benlovell) [inline test/integration/super_admin_flagging_two_step_verification_test.rb]: It was intentional, to be explicit.

**Comment by fofr** [comment]
Copy looks good to me :+1:

**Pull request merged**